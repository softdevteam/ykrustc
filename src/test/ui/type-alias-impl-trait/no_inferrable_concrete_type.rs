// Issue 52985: user code provides no use case that allows a type alias `impl Trait`
// We now emit a 'could not find defining uses' error

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait, impl_trait_in_bindings))]
//[full_tait]~^ WARN incomplete
//[full_tait]~| WARN incomplete

type Foo = impl Copy; //~ could not find defining uses

// make compiler happy about using 'Foo'
fn bar(x: Foo) -> Foo { x }

fn main() {
    let _: Foo = std::mem::transmute(0u8); //[min_tait]~ ERROR not permitted here
}
