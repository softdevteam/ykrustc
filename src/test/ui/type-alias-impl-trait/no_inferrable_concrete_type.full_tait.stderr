warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/no_inferrable_concrete_type.rs:6:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait, impl_trait_in_bindings))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

warning: the feature `impl_trait_in_bindings` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/no_inferrable_concrete_type.rs:6:55
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait, impl_trait_in_bindings))]
   |                                                       ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #63065 <https://github.com/rust-lang/rust/issues/63065> for more information

error: could not find defining uses
  --> $DIR/no_inferrable_concrete_type.rs:10:12
   |
LL | type Foo = impl Copy;
   |            ^^^^^^^^^

error: aborting due to previous error; 2 warnings emitted

